airbyte-integrations/connectors/source-instagram/source_instagram/client/api.py

```diff
@@ -304,16 +306,23 @@ class MediaInsightsAPI(MediaAPI):
 
     def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         for account in self._api.accounts:
-            media = self._get_media(account["instagram_business_account"], {"limit": self.result_return_limit}, ["media_type"])
+            ig_account = account["instagram_business_account"]
+            media = self._get_media(ig_account, {"limit": self.result_return_limit}, ["media_type"])
             for ig_media in media:
-                yield {
-                    **{
-                        "id": ig_media.get("id"),
-                        "page_id": account["page_id"],
-                        "business_account_id": account["instagram_business_account"].get("id"),
-                    },
-                    **{record.get("name"): record.get("values")[0]["value"] for record in self._get_insights(ig_media)},
-                }
+                try:
+                    yield {
+                        **{
+                            "id": ig_media.get("id"),
+                            "page_id": account["page_id"],
+                            "business_account_id": ig_account.get("id"),
+                        },
+                        **{record.get("name"): record.get("values")[0]["value"] for record in self._get_insights(ig_media)},
+                    }
+                except FacebookRequestError as error:
+                    # An error might occur if the media was posted before the most recent time that
+                    # the user's account was converted to a business account from a personal account
+                    logger.error(f"Insights error for business_account_id {ig_account.get('id')}: {error.body()}")
+                    break
```

Comment:
Shouldn't this be continue instead of break? We want to attempt to sync the remaining media items.

Reply:
We receive all Media starting from the very last one, and if on the next Media we get an Insight error, then I see no reason to make inquiries for each Media further, since they were published even earlier. This will only add the number of requests and extra logs to the console.
I did not find a case in which we could need `continue`, and not `break`.

Reply:
Ok I agree with the code then. But in this case could you write a comment that explains why it's a break instead of a continue?